src/rulesets/Base/mapreduce.jl

```diff
@@ -67,30 +67,55 @@ function rrule(
 end
 
 function rrule(
-    config::RuleConfig{>:HasReverseMode}, ::typeof(sum), f, xs::AbstractArray; dims=:
-)
-    fx_and_pullbacks = map(x->rrule_via_ad(config, f, x), xs)
-    y = sum(first, fx_and_pullbacks; dims=dims)
+    config::RuleConfig{>:HasReverseMode}, ::typeof(sum), f::F, xs::AbstractArray; dims=:
+) where {F}
+    project = ProjectTo(xs)
 
-    pullbacks = last.(fx_and_pullbacks)
+    if _uses_input_only(f, eltype(xs))
+        # Then we can compute the forward pass as usual, save nothing but `xs`:
+        y = sum(f, xs; dims=dims)
+        function sum_pullback_easy(dy)
+            dxs = unthunk(dy) .* conj.(only.(only.(derivatives_given_output.(nothing, f, xs))))
```

Comment:
should we have something like `_siso_derivatives_given_output(f, x) = only(only(nothing, f, x)`
so this can be

```suggestion
            dxs = unthunk(dy) .* conj.(_siso_derivatives_given_output.(f, xs))))
```

Reply:
Or `only_derivative_given_output`. But not sure it's worth the extra complication of one more function to track down & know about.

Reply:
I made this a do-block broadcast like the other cases, now, as I think that's more readable.